addons/product/static/src/product_catalog/search/search_panel.js

```diff
@@ -1,54 +1,187 @@
+
+        useBus(this.env.searchModel, "section-line-count-change", this.updateSectionLineCount);
+
+        onWillStart(async () =>  await this.loadSections());
+    }
+
+    get isNonMrpOrderActive() {
+        return (
+            this.env.model.config.context.product_catalog_order_model !== 'mrp.production'
+            && this.env.model.config.context.product_catalog_order_id
+        );
+    }
+
+    get selectedSection() {
+        return this.env.searchModel.selectedSection;
     }
 
     updateActiveValues() {
         super.updateActiveValues();
-        this.state.sectionOfAttributes = this.buildSection();
+        this.state.sectionOfTags = this.buildSectionOfTags();
     }
 
-    buildSection() {
+    buildSectionOfTags() {
         const values = this.env.searchModel.filters[0].values;
         let sections = new Map();
 
         values.forEach(element => {
             const name = element.display_name;
             const id = element.id;
             const count = element.__count;
-
             if (sections.has(name)) {
-                let currentAttr = sections.get(name);
-                currentAttr.get('ids').push(id);
-                currentAttr.set('count', currentAttr.get('count') + count);
+                let currentTag = sections.get(name);
+                currentTag.get('ids').push(id);
+                currentTag.set('count', currentTag.get('count') + count);
             } else if (count > 0) {
-                let newAttr = new Map();
-                newAttr.set('ids', [id]);
-                newAttr.set('count', count);
-                sections.set(name, newAttr);
+                let newTag = new Map();
+                newTag.set('ids', [id]);
+                newTag.set('count', count);
+                sections.set(name, newTag);
             }
         });
 
         return sections;
     }
 
-    toggleSectionFilterValue(filterId, attrIds, { currentTarget }) {
-        attrIds.forEach(id => {
+    toggleTagFilterValue(filterId, tagIds, { currentTarget }) {
+        tagIds.forEach(id => {
             this.toggleFilterValue(filterId, id, { currentTarget });
         })
     }
+
+    onDragStart(sectionId, ev) {
+        ev.dataTransfer.setData('text/plain', sectionId);
+    }
+
+    onDragOver(ev) {
+        ev.preventDefault();
+    }
+
+    onDrop(targetSecId, ev) {
+        ev.preventDefault();
+        const moveSecId = ev.dataTransfer.getData('text/plain');
+        if (moveSecId !== targetSecId) this.reorderSections(moveSecId, targetSecId);
+    }
+
+    enableSectionInput() {
+        this.state.isAddingSection = true;
+        setTimeout(() => document.querySelector('.o_section_input')?.focus(), 100);
+    }
+
+    onSectionInputKeydown(event) {
+        const hotkey = getActiveHotkey(ev);
+        if (hotkey === 'enter') {
+            this.createSection();
+        } else if (hotkey === 'escape') {
+            this.state.isAddingSection = false;
+            this.state.newSectionName = "";
```

Comment:
```suggestion
            Object.assign(this.state, {
                isAddingSection: false,
                newSectionName: "",
            });
```
you can maybe try this.